src/config.ts

```diff
@@ -222,7 +227,42 @@ export const ALLOW_PAGOPA_IP_SOURCE_RANGE = decodeCIDRs(
 export const API_KEY = getRequiredENVVar("API_KEY");
 export const API_URL = getRequiredENVVar("API_URL");
 
-export const API_CLIENT = new ApiClientFactory(API_KEY, API_URL);
+// whether we want to reuse sockets in the fetch client
+const isFetchKeepaliveEnabled = process.env.FETCH_KEEPALIVE_ENABLED === "true";
+
+// Returns a fetch instance backed by a keepalive-enabled HTTP agent
+const getKeepaliveFetch: () => typeof fetch = () => {
+  // see https://nodejs.org/docs/latest-v10.x/api/http.html#http_new_agent_options
+  const keepAliveMaxSockets =
+    process.env.FETCH_KEEPALIVE_MAX_SOCKETS === undefined
+      ? Infinity
+      : parseInt(process.env.FETCH_KEEPALIVE_MAX_SOCKETS, 10);
```

Comment:
I'd add this reference in a comment: https://blog.botframework.com/2018/03/05/fix-snat-exhaustion-node-js-bots/

see also this thread on using the library agentkeepalive (that has a ttl) vs native nodejs keepalive (which seems to work on newer version of node)

https://github.com/MicrosoftDocs/azure-docs/issues/29600#issuecomment-490354812